### PR TITLE
[codegen] Escape variable delimiters in raw comment printing

### PR DESCRIPTION
**Pull request by drfloob**
Fixes https://github.com/grpc/grpc/issues/32171

### CODE REVIEW

File: src/compiler/cpp_generator.cc [hunk 1]
```diff
@@ -54,6 +54,16 @@ std::string FilenameIdentifier(const std::string& filename) {
   }
   return result;
 }
+
+std::string EscapeVariableDelimiters(std::string&& mut_str) {
+  size_t index = 0;
```
Comment (markdroth) [inline src/compiler/cpp_generator.cc:59]: Consider instead using `absl::StrReplaceAll()` for this.

**Review by markdroth — APPROVED** [review]

File: src/compiler/cpp_generator.cc (hunk 1 above)
Comment (drfloob) [reply src/compiler/cpp_generator.cc:59]: That was my first instinct, but the `grpc_plugin_support` target has very few dependencies, and I'm not entirely sure why, it seems almost by design. Maybe for code bloat reasons? Anyhow, that's the only reason I didn't use abseil here.

Comment (veblush) [reply src/compiler/cpp_generator.cc:59]: I feel like we should be careful on adding more dependency to `grpc_plugin_support`.

**Review by veblush — APPROVED** [review]
Thank you for having this quick fix.

**Pull request merged**